tokio/src/task/task_local.rs

```diff
@@ -234,6 +238,8 @@ pin_project! {
         slot: Option<T>,
         #[pin]
         future: F,
+        #[pin]
+        _pinned: PhantomPinned,
```

Comment:
If we don't do this, then #3273 changes whether `TaskLocal::scope` returns an `Unpin` future.

Reply:
This is a good catch :+1: